.github/workflows/run-tpch.yaml

```diff
@@ -0,0 +1,164 @@
+name: Run tpch benchmarks
+
+on:
+  workflow_dispatch:
+    inputs:
+      wheel:
+        type: string
+        description: The wheel artifact to use
+        required: false
+        default: getdaft-0.3.0.dev0-cp38-abi3-manylinux_2_31_x86_64.whl
+      skip_questions:
+        type: string
+        description: The TPC-H questions to skip
+        required: false
+        default: ""
+      scale_factor:
+        type: choice
+        options:
+        - '2'
+        - '10'
+        - '100'
+        - '1000'
+        description: Which scale factor to use
+        required: false
+        default: '2'
+      partition_size:
+        type: choice
+        options:
+        - '2'
+        - '32'
+        - '32'
+        - '100'
+        - '300'
+        - '320'
+        - '512'
+        description: Which partition size to use
+        required: false
+        default: '2'
+      python_version:
+        type: string
+        description: The version of python to use
+        required: false
+        default: "3.9"
+  workflow_call:
+    inputs:
+      wheel:
+        type: string
+        description: The wheel artifact to use
+        required: false
+        default: getdaft-0.3.0.dev0-cp38-abi3-manylinux_2_31_x86_64.whl
+      skip_questions:
+        type: string
+        description: The TPC-H questions to skip
+        required: false
+        default: ""
+      scale_factor:
+        type: string
+        description: Which scale factor to use
+        required: false
+        default: '2'
+      partition_size:
+        type: string
+        description: Which partition size to use
+        required: false
+        default: '2'
+      python_version:
+        type: string
+        description: The version of python to use
+        required: false
+        default: "3.9"
+
+jobs:
+  run-tpch:
+    runs-on: [self-hosted, linux, x64, ci-dev]
+    timeout-minutes: 15 # Remove for ssh debugging
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 1
+    - uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: us-west-2
+        role-session-name: run-tpch-workflow
+    - uses: ./.github/actions/install
+    - run: |
```

Comment:
Split this into steps, this makes it difficult to debug if something goes wrong.

You probably also want the `ray down` step to "always run" regardless of success/failure of all the rest of the steps.